store/sqlstore/channel_store.go

```diff
@@ -1469,3 +1471,61 @@ func (s SqlChannelStore) GetMembersByIds(channelId string, userIds []string) sto
 		}
 	})
 }
+
+// GetChannelsWithChannelContextPermission retrieves the channels for a given user with an associated channel-context of a permission.
+func (s SqlChannelStore) GetChannelsWithChannelContextPermission(userID string, permissionName string) store.StoreChannel {
+	var query string
+
+	if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		query = getChannelsWithChannelContextPermissionPostgresQuery(permissionName)
+	} else {
+		// TODO, Phase 3 Advanced Permissions: role count will need to be queried with something like `select count(*) from roles` because of custom roles.
+		query = getChannelsWithChannelContextPermissionMysqlQuery(permissionName, TOTAL_ROLE_COUNT)
+	}
+
+	return store.Do(func(result *store.StoreResult) {
+		var channels model.ChannelList
+		_, err := s.GetReplica().Select(&channels, query, map[string]interface{}{"UserId": userID})
+		if err != nil {
+			fmt.Printf("err: %v", err)
+			result.Err = model.NewAppError("SqlChannelStore.GetChannelsWithChannelContextPermission", "store.sql_channel.search.app_error", nil, "", http.StatusInternalServerError)
+		} else {
+			result.Data = &channels
+		}
+	})
+}
+
+func getChannelsWithChannelContextPermissionMysqlQuery(permissionName string, roleCount int) string {
+	query := `
+			select Channels.* from (
+			select
+			ChannelId, UserId, trim(substring_index(substring_index(B.Roles, ' ', ns.n), ' ', -1)) as rolename
+			from (
+			select 1 as n union all `
```

Comment:
Can we use the same capitalization we do for other queries? e.g. 'SELECT Channels.* From ...`